packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -24,11 +24,27 @@ export class RemoteMessageProcessor {
 
     public process(remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage {
         let message = copy(remoteMessage);
-
-        message = this.opDecompressor.processMessage(message);
+        message = this.opDecompressor.processMessage(message).message;
         unpackRuntimeMessage(message);
-        message = this.opSplitter.processRemoteMessage(message);
 
+        const chunkProcessingResult = this.opSplitter.processRemoteMessage(message);
+        message = chunkProcessingResult.message;
+        if (chunkProcessingResult.state !== "Processed") {
+            // If the message is not chunked or if the splitter is still rebuilding the original message,
+            // there is no need to continue processing
+            return message;
+        }
+
+        const decompressionAfterChunking = this.opDecompressor.processMessage(message);
```

Comment:
"Processed" above seems like wrong term to use. In a sense that decompression may run on chunked and non-chunked results, so really the code above is conveying that compression should be involved. I think it should talk in that language, as developer changing code in chunking code may not realize what "Processed" means and what it impacts.

Maybe other way to say it - I think code is wrong. Decompression should be running even if "Skipped" was returned because we should totally support batches that fit into single op after compression

Reply:
> Maybe other way to say it - I think code is wrong. Decompression should be running even if "Skipped" was returned because we should totally support batches that fit into single op after compression

We run decompression before we even check the chunk status (see line 27)